platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/EncryptionService.kt

```diff
@@ -107,7 +101,8 @@ class DefaultEncryptionService(
     }
 
     override val encryptionAlgorithm: EncryptionAlgorithm by lazy {
-        SupportedEncryptionAlgorithm.byTransformation(startParameter.encryptionAlgorithm)
+        SupportedEncryptionAlgorithm.getAll().find { it.transformation == startParameter.encryptionAlgorithm }
+            ?: throw InvalidUserDataException("Unsupported encryption algorithm: ${startParameter.encryptionAlgorithm}")
```

Comment:
💭 It would be nice to also list the supported algorithms in the exception message.